### PR TITLE
Add Cite and CiteGroup types

### PR DESCRIPTION
**Pull request by beneboy**
As per title.

### CODE REVIEW

File: schema/Cite.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+| title                 | `string`                                                                        | The title of the creative work.                                                                                         | [CreativeWork](./CreativeWork.html) |
+| url                   | `string<uri>`                                                                   | The URL of the item.                                                                                                    | [Thing](./Thing.html)               |
+| version               | `string` \| `number`                                                            | The version of the creative work.                                                                                       | [CreativeWork](./CreativeWork.html) |
+
+:::
+
+A `Cite` node is used within a [`CreativeWork`](./CreativeWork.html), usually an [`Article`](./Article.html), to refer to an other `CreativeWork`. Often a `Cite` will be associated with other citations, in a [`CiteGroup`](../CiteGroup.html).
```
Comment (alex-ketch) [inline schema/Cite.md]: I think the path is incorrect

```suggestion
A `Cite` node is used within a [`CreativeWork`](./CreativeWork.html), usually an [`Article`](./Article.html), to refer to an other `CreativeWork`. Often a `Cite` will be associated with other citations, in a [`CiteGroup`](./CiteGroup.html).
```

File: schema/Cite.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+# Examples
+
+The following example a (very) short article, shows how a `Cite` is used to refer to another `Article` in the article's `references`.
```
Comment (alex-ketch) [inline schema/Cite.md]:
```suggestion
The following example of a (very) short article, shows how a `Cite` is used to refer to another `Article` in the article's `references`.
```

File: schema/InlineContent.schema.yaml [hunk 3]
```diff
@@ -20,3 +20,4 @@ anyOf:
   - $ref: Strong
   - $ref: Subscript
   - $ref: Superscript
+  - $ref: CiteGroup
```
Comment (alex-ketch) [inline schema/InlineContent.schema.yaml]: Based on the example in `Cite.md` and the HTML [`cite` tag](url) believe `Cite` should also be added to InlineContent here, as well as the typeMap.
```suggestion
  - $ref: Cite
  - $ref: CiteGroup
```

File: tests/__file_snapshots__/Thing.ts [hunk 4]
```diff
@@ -2,7 +2,7 @@
  * The most generic type of item.
  */
 export interface Thing extends Entity {
-  type: 'Thing' | 'Article' | 'AudioObject' | 'Brand' | 'Code' | 'CodeBlock' | 'CodeChunk' | 'CodeExpr' | 'Collection' | 'ContactPoint' | 'CreativeWork' | 'Datatable' | 'DatatableColumn' | 'Environment' | 'ImageObject' | 'MediaObject' | 'Mount' | 'Organization' | 'Periodical' | 'Person' | 'Product' | 'PublicationIssue' | 'PublicationVolume' | 'ResourceParameters' | 'SoftwareApplication' | 'SoftwareSession' | 'SoftwareSourceCode' | 'Table' | 'VideoObject'
+  type: 'Thing' | 'Article' | 'AudioObject' | 'Brand' | 'Cite' | 'Code' | 'CodeBlock' | 'CodeChunk' | 'CodeExpr' | 'Collection' | 'ContactPoint' | 'CreativeWork' | 'Datatable' | 'DatatableColumn' | 'Environment' | 'ImageObject' | 'MediaObject' | 'Mount' | 'Organization' | 'Periodical' | 'Person' | 'Product' | 'PublicationIssue' | 'PublicationVolume' | 'ResourceParameters' | 'SoftwareApplication' | 'SoftwareSession' | 'SoftwareSourceCode' | 'Table' | 'VideoObject'
```
Comment (alex-ketch) [inline tests/__file_snapshots__/Thing.ts]: Not 100% on this, but does `CiteGroup` need to be added here as well?

File: schema/InlineContent.schema.yaml (hunk 3 above)
Comment (beneboy) [reply schema/InlineContent.schema.yaml]: Done in 0113559.

File: schema/Cite.schema.yaml [hunk 5]
```diff
@@ -0,0 +1,47 @@
+title: Cite
+'@id': stencila:Cite
+extends: CreativeWork
```
Comment (nokome) [inline schema/Cite.schema.yaml]: Should extend `Entity`.

**Pull request merged**